### PR TITLE
Merge external and types into a single module

### PR DESCRIPTION
**Pull request by AsafEitani**
* merge external and types into a single module - addresses #1312 
* change external.Event to external.TraceeEvent - as a preparation to address to #1431 

This PR will follow with another PR with the necessary modifications to `tracee/go.mod` and the changes to the code - mainly imports and change external.Event to external.TraceeEvent

### CODE REVIEW

**Review by mtcherni95 — COMMENTED** [review]
_"This PR will follow with another PR with the necessary modifications to tracee/go.mod and the changes to the code - mainly imports and change external.Event to external.TraceeEvent"_ are you referring to this? https://github.com/aquasecurity/tracee/pull/1455

File: external/external.go [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+			tmp[i] = fmt.Sprint(v)
+		}
+		arg.Value = tmp
+	}
+	return nil
+}
+
+// SlimCred struct is a slim version of the kernel's cred struct
+// it is used to unmarshal binary data and therefore should match (bit by bit) to the `slim_cred_t` struct in the ebpf code.
+// ANY CHANGE TO THIS STRUCT WILL BE REQUIRED ALSO TO bufferdecoder.SlimCred
```
Comment (mtcherni95) [inline external/external.go:198]: please update here https://github.com/aquasecurity/tracee/blob/main/tracee-ebpf/tracee/internal/bufferdecoder/protocol.go#L91 that changes must be done to both `SlimCred` structs: both in `external/external.go` and in `pkg/external/external.go`. 
Are we planning eventually to remove `pkg/external` with this https://github.com/aquasecurity/tracee/pull/1455?

Comment (AsafEitani) [reply external/external.go:198]: Yes we will merge those two one after the other.
I removed the old external in #1455 thanks for noticing.

**Comment by AsafEitani** [comment]
> _"This PR will follow with another PR with the necessary modifications to tracee/go.mod and the changes to the code - mainly imports and change external.Event to external.TraceeEvent"_ are you referring to this? #1455

Yes

File: tracee-ebpf/tracee/internal/bufferdecoder/decoder_test.go [hunk 2]
```diff
@@ -707,7 +707,7 @@ func BenchmarkBinaryBool(*testing.B) {
 
 func BenchmarkDecodeSlimCred(*testing.B) {
 	/*
-		s := external.SlimCred{
+		s := types.SlimCred{
```
Comment (mtcherni95) [inline tracee-ebpf/tracee/internal/bufferdecoder/decoder_test.go:710]: this should actually be `bufferdecoder.SlimCred`

File: tracee-ebpf/tracee/internal/bufferdecoder/decoder_test.go [hunk 3]
```diff
@@ -741,7 +741,7 @@ func BenchmarkDecodeSlimCred(*testing.B) {
 
 func BenchmarkBinarySlimCred(*testing.B) {
 	/*
-		s := external.SlimCred{
+		s := types.SlimCred{
```
Comment (mtcherni95) [inline tracee-ebpf/tracee/internal/bufferdecoder/decoder_test.go:744]: ditto

File: tracee-ebpf/tracee/internal/bufferdecoder/protocol.go [hunk 4]
```diff
@@ -88,7 +88,7 @@ func (MprotectWriteMeta) GetSizeBytes() uint32 {
 
 // SlimCred struct is a slim version of the kernel's cred struct
 // it is used to unmarshal binary data and therefore should match (bit by bit) to the `slim_cred_t` struct in the ebpf code.
-// ANY CHANGE TO THIS STRUCT WILL BE REQUIRED ALSO TO external.SlimCred
+// ANY CHANGE TO THIS STRUCT WILL BE REQUIRED ALSO TO types.SlimCred
```
Comment (mtcherni95) [inline tracee-ebpf/tracee/internal/bufferdecoder/protocol.go:91]: `// ANY CHANGE TO THIS STRUCT WILL BE REQUIRED ALSO TO types.SlimCred and bufferdecoder.SlimCred`

**Review by mtcherni95 — CHANGES_REQUESTED** [review]

File: pkg/procinfo/procinfo.go [hunk 5]
```diff
@@ -34,26 +34,31 @@ type ProcInfo struct {
 }
 
 func (p *ProcInfo) UpdateElement(hostTid int, ctx ProcessCtx) {
-	p.mtx.RLock()
+	p.mtx.Lock()
```
Comment (mtcherni95) [inline pkg/procinfo/procinfo.go:37]: I think you had some issues in rebasing. I remember committing these changes.

File: pkg/types/go.mod [hunk 6]
```diff
@@ -0,0 +1,11 @@
+module github.com/aquasecurity/tracee/pkg/types
```
Comment (danielpacak) [inline pkg/types/go.mod:1]: Do we really need to move it under pkg/types/? The pkg in this Go repo convention is for shared code that you can import as library. The types modules was introduced solely as a workaround to make Go plugins happy so I'd not mix it with code under pkg.

**Review by danielpacak — COMMENTED** [review]
Overall LGTM. I'm just wondering why we move this module under pkg?

File: pkg/types/go.mod (hunk 6 above)
Comment (AsafEitani) [reply pkg/types/go.mod:1]: It was an idea we had but I will move it to tracee/types

**Review by mtcherni95 — COMMENTED** [review]
LGTM!

**Review by mtcherni95 — APPROVED** [review]

**Comment by danielpacak** [comment]
Some Test CO-RE is flaky for know reasons. I've tested changes in my VM and the TRC-2 detection is captured properly.

**Pull request merged**